frontend/rust-lib/flowy-document2/src/parser/json/parser.rs

```diff
@@ -69,14 +91,35 @@ impl JsonToDocumentParser {
     children_map
   }
 
-  fn block_to_block_pb(block: &Block, id: Option<String>, parent_id: String) -> BlockPB {
+  fn block_to_block_pb(
+    block: &Block,
+    id: Option<String>,
+    parent_id: String,
+  ) -> (BlockPB, Option<String>) {
     let id = id.unwrap_or_else(|| nanoid!(10));
-    BlockPB {
-      id,
-      ty: block.ty.clone(),
-      data: serde_json::to_string(&block.data).unwrap(),
-      parent_id,
-      children_id: nanoid!(10),
-    }
+    let mut data = block.data.clone();
+
+    let delta: Option<String> = data.get("delta").map(|d| d.to_string());
```

Comment:
also, you can use `remove` instead of `get` here and delete the `remove` below.

let delta = data.remove.map;